### ISSUE TITLE
error compiling nested CALL-WITH-VALUES uses with other stuff

### ISSUE THREAD

**Issue by marcomaggi** [cisco/ChezScheme#208]
On my `x86_64-slackware-linux-gnu` running a Chez Scheme checkout with latest commit `93a5d4fbfc3c9d77957e6be1e14a88f0365720fe` the program below fails with:

```
Exception in np-insert-trap-check: unexpected Triv with irritant #[#{L10.5:literal:Triv.798 b1zvm1k567difqgipxhkbk-5781} 12 #<literal list>]
```

This is the program, sorry but I am unable to make it smaller than this:

```scheme
#!r6rs
(import (chezscheme))

(define-syntax begin0
  (syntax-rules ()
    ((_ ?form0 ?form ...)
     (call-with-values
	 (lambda () ?form0)
       (lambda args
	 (begin ?form ...)
	 (apply values args))))))

(define result
  (make-parameter #f))

(define-syntax with-result
  (syntax-rules ()
    ((_ ?form ... ?last-form)
     (parameterize ((result '()))
       ?form ...
       (call-with-values
	   (lambda () ?last-form)
	 (lambda args
	   (append args (list (get-result)))))))))

(define (add-result value)
  (result (cons value (result)))
  value)

(define (get-result)
  (reverse (result)))

(with-result
  (call-with-values
      (lambda ()
	(begin0
	    (values 1 10)
	  (add-result 2)
	  (add-result 3)))
    list))

;;; end of file
```

**Comment by yjqww6** [comment]
I reproduced it with shorter code
```
Chez Scheme Version 9.4.1
Copyright 1984-2017 Cisco Systems, Inc.

> (call-with-values
    (lambda ()
      (call-with-values
        (lambda ()
          (+ 1 (current-output-port)))
        list))
    (lambda l l))
Exception in np-insert-trap-check: unexpected Triv with irritant #[#{L10.5:literal:Triv.798 a36te8sh9ibmusdxz4ign3vuvtly0qjt-5784} 12 #<literal list>]
Type (debug) to enter the debugger.
>
```

**Comment by yjqww6** [comment]
in np-normalize-context
https://github.com/cisco/ChezScheme/blob/e00e8a113cfdc1d6294e1f36fa46e3ee715ea517/s/cpnanopass.ss#L9829
Should it be `[(mvset ,info (,mdcl ,[t0?] ,[t1] ...) (,[t*] ...) ((,x** ...) ,interface* ,l*) ...)` ?
After changed this line, I could get code above compiled.

**Comment by akeep** [comment]
@yjqww6 Yep, that is definitely a problem.  I'm going to take a look and see if there are any other problems in ``np-normalize-context`` then I can commit a fix... unless you'd like make a pull request for this and get credit (where it is due) for finding the problem.

**Comment by yjqww6** [comment]
@akeep Please fix it. It's not convenient for me to make a pull request now.

**Comment by akeep** [comment]
@yjqww6 done!

Fixed by 28f31d8.
